api/envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -0,0 +1,142 @@
+// * Service account "cluster.local/ns/default/sa/admin" has full access to the service.
+// * Any user can read ("GET") the service at paths with prefix "/products" or suffix "/reviews" when request header
+//   "version" set to either "v1" or "v2".
+//
+//   action_type: ALLOW
+//   policies:
+//     "service-admin":
+//       permissions:
+//       - methods: ["*"]
+//       principals:
+//       - name: "cluster.local/ns/default/sa/admin"
+//     "product-viewer":
+//       permissions:
+//       - paths: [prefix:"/products", suffix:"/reviews"]
+//         methods: ["GET"]
+//         conditions:
+//         - header:
+//             key: "version"
+//             values: ["v1", "v2"]
+//       principals:
+//       - name: "*"
+//
+message RBAC {
+  // Should we do white-list or black-list style access control.
+  enum ActionType {
+    // The policies grant access to principals. The rest is denied. This is white-list style
+    // access control. This is the default type.
+    ALLOW = 0;
+
+    // The policies deny access to principals. The rest is allowed. This is black-list style
+    // access control.
+    DENY = 1;
+  }
+
+  ActionType action_type = 1;
+
+  // Maps from policy name to policy.
+  map<string, Policy> policies = 2;
+}
+
+// Policy specifies a role and the principals that are assigned/denied the role.
+message Policy {
+ // Required. The set of permissions that define a role.
+ repeated Permission permissions = 1 [(validate.rules).repeated .min_items = 1];
+
+ // Required. List of principals that are assigned/denied the role based on “action_type”.
+ repeated Principal principals = 2 [(validate.rules).repeated .min_items = 1];
+}
+
+message StringMatch {
+  oneof match_pattern {
+    // The input string must match exactly the string specified here.
+    // Or it is a "*", which means that it matches any string.
+    string simple = 1;
+
+    // The input string must have the prefix specified here.
+    string prefix = 2;
+
+    // The input string must have the suffix specified here.
+    string suffix = 3;
+
+    // The input string must match the regular expression specified here.
+    // The regex grammar is defined `here
+    // <http://en.cppreference.com/w/cpp/regex/ecmascript>`_.
+    string regex = 4;
+ }
+}
+
+// Specifies how to match an entry in a map.
+message MapEntryMatch {
+  // The key to select an entry from the map.
+  string key = 1;
+
+  // A list of matched values.
+  repeated StringMatch values = 2;
+}
+
+// Permission defines a permission to access the service.
+message Permission {
+  // Optional. A list of HTTP paths or gRPC methods.
+  // gRPC methods must be presented as fully-qualified name in the form of
+  // packageName.serviceName/methodName.
+  // If this field is unset, it applies to any path.
+  repeated StringMatch paths = 1;
+
+  // Required. A list of HTTP methods (e.g., "GET", "POST").
+  // If set to ["*"], it applies to any method.
```

Comment:
Since this is repeated it seems like potentially an empty list would be better than '*' for any method? WDYT?

Reply:
I made "methods" a required field. Suppose I need to specify "full access" permission, what I showed in the example policy is:
```
   permissions:
   - methods: ["*"]
```
What you mentioned is another alternative. With that, you would write the following to express the same:
```
   permissions:
   - 
```
These are just two different ways to express the policy. Either way is fine. I have slight preference towards the former. WDYT?

Reply:
I prefer the latter. The reason being that otherwise you need special case error handling logic if the user does something like `["*", "GET"]`. I would just have empty be all methods allowed. But we can have other folks weigh in to see what they think. (In general remember that we are not optimizing for human consumption here. We are optimizing for machine generation and consumption.)

Reply:
SGTM. As you said, it is for machine consumption, not for human. For that, consistency is more important.

Reply:
+1 to empty list